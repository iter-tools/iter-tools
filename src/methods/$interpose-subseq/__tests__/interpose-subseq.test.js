/**
 * @generated-from ./$interpose-subseq.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

/* eslint-disable no-unused-vars,import/no-duplicates,no-constant-condition */

import { interposeSubseq, toArray, range } from '../../..';

describe('interposeSubseq', () => {
  it('interposes items into array', () => {
    const iter = interposeSubseq([9, 9], [1, 2, 3]);
    expect(toArray(iter)).toEqual([1, 9, 9, 2, 9, 9, 3]);
  });

  it('interposes items into an iterable', () => {
    const iter = interposeSubseq([null], range({ start: 1, end: 4 }));
    expect(toArray(iter)).toEqual([1, null, 2, null, 3]);
  });

  it('returns mapped iterable (curried version)', () => {
    const iter = interposeSubseq([]);
    expect(toArray(iter(range({ start: 1, end: 4 })))).toEqual([1, 2, 3]);
  });

  it('returns empty iterable from null', () => {
    expect(toArray(interposeSubseq('', null))).toEqual([]);
  });
});
