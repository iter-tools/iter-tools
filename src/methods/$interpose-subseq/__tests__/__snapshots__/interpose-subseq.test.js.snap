// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interposeSubseq when source is a string warns 1`] = `Array []`;
